mesonbuild/modules/pkgconfig.py

```diff
@@ -297,22 +307,42 @@ def generate_libs_flags(libs):
                         if 'cs' not in l.compilers:
                             yield '-l%s' % lname
 
+            def generate_include_flags(libs):
```

Comment:
this name is confusing. I think it should be `generate_uninstalled_include_flags`.
Also it seems to do something quite different than the installed case.

The paths used for compilation of a library don't need to match what the library user expects. For one they might include tons of private things. And there are quite some libraries that get compiled with paths pointing inside the library specific include directory instead of pointing only to the directory containing it.

Of course the includes in the uninstalled case are a hard to get in the general case. I think the only really supportable case is when subdirs points to directories that are 1:1 (subdir matching the last elements of the path of all files) installed and where nothing else is placed into by other install related calls. I'm not sure how generated header fit into this, but i think there should be a way to allow them too.

Reply:
Right, renamed that function. I'll reply below about the include directories.